### DISCUSSION TITLE
Is it possible in DVC to pull and push only subfiles within a .dvc file ?

### DISCUSSION THREAD

**Discussion by aveek-brahmachari** [treeverse/dvc#6818] [Help] [answered]
Suppose, I have a folder XYZ that has subfolders and data ABC and DEF. Is there anyway to push and pull only ABC using XYZ.dvc ?

**Comment by efiop** [comment, marked as answer]
Hi @aveek-brahmachari . Yes, you can `dvc push/pull XYZ/ABC`

**Reply by nvinayvarma189** [reply]
Hi, @efiop I can't seem to get this working.

I have a folder called `data` with a subfolder (which again has multiple subfolders). I also do have a `data.dvc` file and have pushed the `data` folder to S3. 

Now when I do `dvc pull` from another instance of the same repo, I get the data folder downloaded from S3 but if I do `dvc pull data/subfolder` I get this message

```
Everything is up to date.
```

Can you please help me here? Or if "pushing/pulling subfolders" is documented somewhere, can you please point me to it?

**Reply by efiop** [reply]
@nvinayvarma189 Does `data/subfolder` exist after `dvc pull`?

**Reply by nvinayvarma189** [reply]
Do you mean in the place where I executed `dvc pull`? No.

**Reply by efiop** [reply]
@nvinayvarma189 Could you check `dvc list . data/subfolder`, does it show anything?

**Reply by nvinayvarma189** [reply]
I get the following when I execute `dvc list . data/subfolder`

```
ERROR: unexpected error - ('/', 'root', 'vinay', 'dummy', 'dvc_test', 'data', 'subfolder')
```
output of `pwd` is
```
/root/vinay/dummy/dvc_test
```

I'm using `Version: 2.8.1` btw.

**Reply by efiop** [reply]
@nvinayvarma189 How about `dvc list . data`? Or `dvc list .`(run this in the repo root, btw). If you don't see `data` at all in its output - most likely you didn't `git commit/push` corresponding meta files in the original machine.

**Reply by nvinayvarma189** [reply]
I'd like to apologize for an overlook mistake from my end, `dvc pull data/subfolder` workers perfectly. I hope `dvc push data/subfolder` also works nicely :)

Thank you very much :pray:
